swagger_server/dal/__init__.py

```diff
@@ -0,0 +1,37 @@
+from flask import current_app
+
+from swagger_server.helpers import db
+
+
+def get_nearest_neighbours(sample_id):
+    with db.Neo4jDatabase.get() as db_helper:
```

Comment:
Fixed: Start & close each connection for each request instead (see the how the `helpers.controller_helpers.with_db` is used in each controller method)

Reply:
This is not much different to starting and closing db connections per query because most of requests will require only one db query. If you look into the details of the Python driver implementation, it holds a connection pool. So we think we should re-use the driver between requests.

Reply:
Fixed @Zhicheng-Liu